### DISCUSSION TITLE
MessageBox.Avalonia auto-select TabItem if bound

### DISCUSSION THREAD

**Discussion by giacarrea** [AvaloniaUI/Avalonia#13907] [Q&A] [answered]
Hello, I'm having an issue between my TabControl and MessageBox.Avalonia; 

My TabControl has two TabItems with bound "IsSelected" properties. Whenever IMsBox.ShowAsPopupAsync is called, my TabControl automatically selets the first tab.

This only happens when the IsSelected properties are bound, even if the bound properties are not used elsewhere. 

Has anyone had similar issues?

**Comment by giacarrea** [comment]
Here's an app - derived from the avalonia sample project - reproducing the issue:
[Avalonia.Samples-main.zip](https://github.com/AvaloniaUI/Avalonia/files/13635956/Avalonia.Samples-main.zip)

Start the BasicMvvmSample project, then in the "PlotView" tab, click "Open window" -> "Panel2", then click "Update device config".

**Reply by timunie** [reply]
do you use CompiledBindings? If not, enable them to debug binding issues.

**Reply by giacarrea** [reply]
I enabled them, but couldn't see anything in particular. 

Although I did noticed the following lines appearing whenever I clicked my button in the Debug output window (regardless of whether or not the compiled bindings were enabled):

[Binding]Error in binding to 'Avalonia.Input.KeyBinding'.'Command': 'Null value in expression '{empty}' at ''.'(KeyBinding #45048586)
[Binding]Error in binding to 'Avalonia.Input.KeyBinding'.'Command': 'Null value in expression '{empty}' at ''.'(KeyBinding #2784095)
[Binding]Error in binding to 'Avalonia.Input.KeyBinding'.'Command': 'Null value in expression '{empty}' at ''.'(KeyBinding #25056856)

**Reply by timunie** [reply]
Sorry, but I cannot debug your entire App. That's too much code for me to review. Please extract a very basic sample to reproduce. 

> [!NOTE] 
> The Binding messages are known and can be ignored. We're working on a solution...

**Reply by giacarrea** [reply]
Here's a project with only the code manifesting the issue remaining:

[Avalonia.Samples-main - messagebox bug only.zip](https://github.com/AvaloniaUI/Avalonia/files/13639556/Avalonia.Samples-main.-.messagebox.bug.only.zip)

As before, just start the BasicMvvmSample project, then click on the button in Tab3, Tab1 should be selected as soon as the dialogbox is displayed.

**Reply by timunie** [reply, marked as answer]
checked just now. I see two possible solutions here:

### 1: Do not use Popup-Dialogs

If I try `bonkDialog.ShowWindowDialogAsync(parentWindow);` it will not change tab selection. Looks like MessageBox Avalonia is manipulating the tree somehow and thus it *may* be a bug on their side. 

### 2: Use SelectedItem or SelectedIndex

I updated the VM like this: 

```cs
private int _TabIndex = 2;
/// <summary>
/// Gets or sets the current TabIndex
/// </summary>
public int TabIndex
{
    get => _TabIndex;
    set => this.RaiseAndSetIfChanged(ref _TabIndex, value);
}
```

and then in your XAML it's simply: 

```xml
<TabControl SelectedIndex="{Binding TabIndex}">
	<TabItem Header="Tab1">
		<Label>Tab1</Label>
	</TabItem>
	<TabItem Header="Tab2">
		<Label>Tab2</Label>
	</TabItem>
	<TabItem Header="Tab3">
		<Button Command="{Binding FakeUpdateDevice}"
		        Content="Open dialog" />
	</TabItem>
</TabControl>
```

> Solution 2 is my favorite tbh.

**Reply by giacarrea** [reply]
Yep, that works, thanks. Is it worth tuning this into a bug ticket?

**Reply by timunie** [reply]
Not sure here tbh. I guess that binding to SelectedIndex is preferred over IsSelected on TabItem. That also adds flexibility if you need dnymic Tabs generated.
